src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2235 @@
+    dst_f8->f1[2] = fmaf(src_f1, filter[7], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter[6], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[5], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[4], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[3], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[2], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack2(src_ui4.z);
+    dst_f8->f1[2] = fmaf(src_f1, filter[8], dst_f8->f1[2]);
+    dst_f8->f1[3] = fmaf(src_f1, filter[7], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[6], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[5], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[4], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[3], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack3(src_ui4.z);
+    dst_f8->f1[3] = fmaf(src_f1, filter[8], dst_f8->f1[3]);
+    dst_f8->f1[4] = fmaf(src_f1, filter[7], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[6], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[5], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[4], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack0(src_ui4.w);
+    dst_f8->f1[4] = fmaf(src_f1, filter[8], dst_f8->f1[4]);
+    dst_f8->f1[5] = fmaf(src_f1, filter[7], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[6], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[5], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack1(src_ui4.w);
+    dst_f8->f1[5] = fmaf(src_f1, filter[8], dst_f8->f1[5]);
+    dst_f8->f1[6] = fmaf(src_f1, filter[7], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[6], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack2(src_ui4.w);
+    dst_f8->f1[6] = fmaf(src_f1, filter[8], dst_f8->f1[6]);
+    dst_f8->f1[7] = fmaf(src_f1, filter[7], dst_f8->f1[7]);
+    src_f1 = rpp_hip_unpack3(src_ui4.w);
+    dst_f8->f1[7] = fmaf(src_f1, filter[8], dst_f8->f1[7]);
+}
+
+// -------------------- Set 1 - PKD3->PKD3 for T = U8/F32/F16/I8 --------------------
+
+// kernelSize = 3
+template <typename T>
+__global__ void gaussian_filter_3x3_pkd_tensor(T *srcPtr,
+                                               uint2 srcStridesNH,
+                                               T *dstPtr,
+                                               uint2 dstStridesNH,
+                                               uint padLength,
+                                               uint2 tileSize,
+                                               RpptROIPtr roiTensorPtrSrc,
+                                               d_float9 *filterTensor)
+{
+    int hipThreadIdx_x8 = hipThreadIdx_x << 3;
+    int id_x_o = (hipBlockIdx_x * tileSize.x * 8) + hipThreadIdx_x8;
+    int id_y_o = hipBlockIdx_y * tileSize.y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    int id_x_i = id_x_o - padLength;
+    int id_y_i = id_y_o - padLength;
+    d_float24 sum_f24;
+    __shared__ uchar src_lds[48][128];
```

Comment:
can you comment here on what these 48 and 128 numbers are? it's better to define these numbers in the header and use the definitions here for better code readability.

Reply:
Added a well explained reasoning now @asalmanp at https://github.com/GPUOpen-ProfessionalCompute-Libraries/rpp/pull/180/commits/c7940a01421984fdc23e4449c3f21759bdd439c0